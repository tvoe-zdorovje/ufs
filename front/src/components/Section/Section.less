@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
