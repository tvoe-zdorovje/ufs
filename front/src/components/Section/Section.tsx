import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
