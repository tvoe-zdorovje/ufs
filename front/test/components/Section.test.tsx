import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'

import Section from 'components/Section/Section'

describe('Компонент <Section />', () => {
  const hi = 'hi'

  const wrapper = shallow(<Section header={hi}><textarea>text</textarea></Section>)
  const renderWrapper = render(<Section header={hi}><textarea /></Section>)

  it('использует свойство header', () => {
    expect(wrapper.find('div').first().text()).toBe(hi)
  })

  it('отображает дочерние компоненты', () => {
    expect(wrapper.find('textarea').text()).toBe('text')
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
