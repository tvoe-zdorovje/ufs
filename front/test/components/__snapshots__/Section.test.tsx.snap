// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <Section /> совпадает со своим снимком 1`] = `
<section
  class="section"
>
  <div
    class="header"
  >
    hi
  </div>
  <div
    class="container"
  >
    <textarea />
  </div>
</section>
`;
